### PR TITLE
Initial port to 2.0

### PR DESCRIPTION
**Pull request by cdata**

### CODE REVIEW

File: iron-control-state.html [hunk 1]
```diff
@@ -51,8 +51,17 @@
...
         }
-      }
+      },
 
+      __handleEventRetargetting: {
+        type: Boolean,
+        value: function() {
+          // NOTE(cdata): Polymer 2.0 hybrid mode should take care of messy
+          // focus event retargetting for us.
+          return !this.shadowRoot &&
```
Comment (notwaldorf) [inline iron-control-state.html]: I think Polymer.Element is better than a regex

**Comment by notwaldorf** [comment]
@bicknellr made changes, 👀 ?

File: iron-control-state.html [hunk 2]
```diff
@@ -65,15 +71,20 @@
     },
 
     _focusBlurHandler: function(event) {
+      // In Polymer 2.0, the library takes care of retargeting events.
```
Comment (bicknellr) [inline iron-control-state.html]: :clap:

File: test/focused-state.html [hunk 3]
```diff
@@ -124,15 +124,15 @@
 
       setup(function() {
         lightDOM = fixture('LightDOM');
-        input = document.querySelector('#input');
+        input = lightDOM.querySelector('#input');
```
Comment (bicknellr) [inline test/focused-state.html]: This should probably also go through `Polymer.dom`, right? There are a few other cases:
- https://github.com/PolymerElements/iron-behaviors/blob/c40e86d9ab2ad2e96d77c0132e1211d900ad1d10/test/active-state.html#L209
- https://github.com/PolymerElements/iron-behaviors/blob/c40e86d9ab2ad2e96d77c0132e1211d900ad1d10/test/active-state.html#L224
- https://github.com/PolymerElements/iron-behaviors/blob/c40e86d9ab2ad2e96d77c0132e1211d900ad1d10/test/active-state.html#L245
- https://github.com/PolymerElements/iron-behaviors/blob/c40e86d9ab2ad2e96d77c0132e1211d900ad1d10/test/active-state.html#L260

File: iron-control-state.html [hunk 4]
```diff
@@ -51,8 +51,14 @@
         value: function() {
           return this._focusBlurHandler.bind(this);
         }
-      }
+      },
 
+      __handleEventRetargetting: {
```
Comment (bicknellr) [inline iron-control-state.html]: `s/targetting/targeting/`

File: test/focused-state.html (hunk 3 above)
Comment (notwaldorf) [inline test/focused-state.html]: No, because it's in the light dom...no?

File: iron-control-state.html (hunk 4 above)
Comment (notwaldorf) [inline iron-control-state.html]: it's not incorrect!!! it's the british spelling lol. LET ME LIVE

File: test/focused-state.html (hunk 3 above)
Comment (bicknellr) [inline test/focused-state.html]: Huh, why does the line below use `Polymer.dom`?

Comment (notwaldorf) [inline test/focused-state.html]: Dunno, I didn't write these tests :)

File: iron-control-state.html (hunk 4 above)
Comment (bicknellr) [inline iron-control-state.html]: https://en.wiktionary.org/wiki/targetting
> 1. (British spelling) Obsolete spelling of targeting
> 2. Misspelling of targeting.

https://books.google.com/ngrams/graph?content=targeting%2Ctargetting&year_start=1900&year_end=2008&corpus=15&smoothing=3

'Muricanet disagrees.

Comment (bicknellr) [inline iron-control-state.html]: (but no, this isn't important :) )

File: test/focused-state.html (hunk 3 above)
Comment (bicknellr) [inline test/focused-state.html]: Found it in the docs: https://www.polymer-project.org/1.0/docs/devguide/local-dom#dom-api . I think this should go through `Polymer.dom`.

File: iron-control-state.html (hunk 4 above)
Comment (notwaldorf) [inline iron-control-state.html]: UGH FINE

File: iron-control-state.html [hunk 5]
```diff
@@ -51,8 +51,14 @@
         value: function() {
           return this._focusBlurHandler.bind(this);
         }
-      }
+      },
 
+      __handleEventRetargeting: {
```
Comment (bicknellr) [inline iron-control-state.html]: :eagle:

File: test/focused-state.html [hunk 6]
```diff
@@ -124,15 +124,15 @@
 
       setup(function() {
         lightDOM = fixture('LightDOM');
-        input = document.querySelector('#input');
+        input = Polymer.dom(lightDOM).querySelector('#input');
```
Comment (bicknellr) [inline test/focused-state.html]: 👍

**Pull request merged**